### ISSUE TITLE
Move away from differentiating between free and paid content messages

### ISSUE THREAD

**Issue by tnorthcutt** [restrictcontentpro/restrict-content-pro#1701]
In #914 it was suggested that we move away from differentiating between free and paid content messages. Because content can be restricted in several different ways, and can be restricted to both free and paid levels, having separate messages for free and paid content can be confusing and inaccurate.

A major factor to consider will be backwards compatibility. How do we make this change, without negatively affecting existing sites that use both free and paid messages?

Options:
1. Wait until a major version bump (3.0, 4.0, etc.), remove the dual messaging at that time, and attempt to notify site owners/operators of the change.
2. Make the change opt-in for existing sites, and on by default for new installs. IMO, this would just kick the can down the road, so to speak.
3. ???

I'm sure there are other options I'm missing right now.

Also worth discussing: are there other places, besides messaging, where we can or should move away from differentiating between free and paid?

**Comment by mindctrl** [comment]
I think this is a good candidate for 3.1. That's effectively a major release with the way we do things, and we currently have a lot going on in 3.0 and I'm hesitant to add to it.

I'd like to announce this ahead of time, so option 1 seems like the best to me. I agree that option 2 just delays the inevitable and adds more cruft to our code.

Regarding things other than messaging: I think we should stick with messaging only, at least in this issue. With content restrictions, I see the value in having content restricted to 'any' sub and also any 'paid' sub. As far as statuses go, we've long disliked the free vs active bit, but that's a larger change related to #774 and we can discuss there.

**Comment by tnorthcutt** [comment]
@mindctrl I like that approach. One thing I should have mentioned above: we should consider how the change will affect existing sites, in that they might go from two customized restricted messages, to using the new default message when they click upgrade.

One (complicated, but site-owner-friendly) approach might be to remove the UI for the old messages, but still check for their existence, and use them if they're set. And, if so, also show a prominent admin notice to get the site owner to save a new "universal" message.

We could leave that in place for a few minor releases to give people time to notice save a new message, then remove the checks and completely rip out the old messaging code in 3.x.

Again, that's obviously more complicated, but perhaps worth the trouble for the user-friendliness? I'm sure people who see RCP support requests regularly would have a better idea than me of if it's worth it 😁

**Comment by nosegraze** [comment]
I would be okay with removing the UI but still check for their existence for a while. That would give more of a grace period for people who don't test new versions before updating.

**Comment by mindctrl** [comment]
> I would be okay with removing the UI but still check for their existence for a while.

I like that.

**Comment by nosegraze** [comment]
I left a few comments on the PR. However, I don't think we should invest too much time in it now since we're thinking of this for `3.1`. A lot can change between now and then and we may want to make more changes to this PR down the line.

**Comment by tnorthcutt** [comment]
@nosegraze roger that. I went ahead and cleaned it up just so it's not hanging out there. Happy to let it sit for now.

**Comment by pippinsplugins** [comment]
Since this is largely done, how about we put it in 3.0?

**Comment by nosegraze** [comment]
Yes this definitely needs to be a major release. It's currently slated for `3.1` but we can look at it for `3.0`.

**Comment by tnorthcutt** [comment]
Sounds good to me. I went ahead and switched the milestone to `3.0` for now.
